### PR TITLE
(SERVER-3157) Add Snyk GH Action

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

**Review by mwaggett — APPROVED** [review]

**Comment by Magisus** [comment]
Will need to update this to `pull_request_target` once we see Snyk go green, before we merge. Also looking into why Travis's unit tests aren't running...

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        fetch-depth: 0
+        persist-credentials: false
+    - name: Run Clojure Snyk Scan
+      id: scan
+      uses: puppetlabs/security-snyk-clojure-action@main
+      with:
+        snykToken: ${{ secrets.SNYK_PE_TOKEN }}
+        rproxyKey: ${{ secrets.SEC_PUBLIC_RPROXY_KEY }}
+        snykOrg: 'puppet-foss'
+        snykProject: 'jruby-deps'
```
Comment (mwaggett) [inline .github/workflows/snyk.yml:25]: should this be `jruby-utils`?

Comment (Magisus) [reply .github/workflows/snyk.yml:25]: Good catch thanks

**Review by mwaggett — APPROVED** [review]

**Review by mwaggett — APPROVED** [review]

**Comment by Magisus** [comment]
I think this is ready now

**Pull request merged**